src/Microsoft.Health.Fhir.Core/Features/Operations/DataConvert/Models/ConvertEngineInitializeException.cs

```diff
@@ -0,0 +1,36 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models
+{
+    public class ConvertEngineInitializeException : FhirException
+    {
+        public ConvertEngineInitializeException(string message)
+            : base(message)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(message, nameof(message));
```

Comment:
Genreally, throwing inside of an exception's constructor is not a great idea, because if this condition occurs, you'll get an ArgumentException and the underlying issue is lost. I would do a Debug.Assert() instead.

Reply:
Thanks. That's a great point. Have updated to Debug.Assert().